### PR TITLE
add 21cmFAST recipe

### PR DESCRIPTION
**Pull request by steven-murray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by steven-murray** [comment]
@conda-forge/staged-recipes this is giving me a linter error, but locally the linter tells me the recipe is "in fine form". How do I figure out what's wrong?

File: recipes/21cmFAST/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
```
Comment (scopatz) [inline recipes/21cmFAST/meta.yaml:5]: Please remove comment lines

File: recipes/21cmFAST/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    - scipy
+    - numpy
+    - click
+    - pyyaml
+    - cffi >=1.0
+    - scipy
+    - astropy >=2.0
+    - h5py >=2.8.0
+    - cached-property
+    - matplotlib
```
Comment (scopatz) [inline recipes/21cmFAST/meta.yaml:60]: Should be matplotlub-base

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: Thanks -- didn't realize that!

**Comment by steven-murray** [comment]
Thanks for the help @scopatz, it looks to be all OK now.

File: recipes/21cmFAST/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 0acf356e74ad8a2254f6c729b06b98f4b6cdde0b838e46c614d9ae68f8e56791
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: Ah, yes. OK this is done now.

File: recipes/21cmFAST/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    - cffi >=1.0
+    - scipy
+    - astropy >=2.0
+    - h5py >=2.8.0
+    - cached-property
+    - matplotlib-base
+    - fftw
+    - gsl
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
```
Comment (isuruf) [inline recipes/21cmFAST/meta.yaml:47]: Please remove these 2 lines

File: recipes/21cmFAST/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  host:
+    - python
+    - pip
+    - cffi
+    - fftw
+    - gsl
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+    - libopenblas
```
Comment (isuruf) [inline recipes/21cmFAST/meta.yaml:31]: Please see docs on how to specify blas.

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml:31]: I'm finding the docs a bit hard to understand here -- my impression is that I should just have `libblas` in the `host` section, and nowhere else. Does that seem right?

Comment (isuruf) [reply recipes/21cmFAST/meta.yaml:31]: Yes, that's right. You also need `libcblas` if the package is using cblas symbols and `liblapack` if the package is using lapack symbols.

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: OK, thanks. I *think* it just needs `blas`, and that is now fixed.

File: recipes/21cmFAST/meta.yaml (hunk 4 above)
Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: Done!

File: recipes/21cmFAST/meta.yaml (hunk 5 above)
Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: All runs OK so it looks like that was correct.

**Comment by steven-murray** [comment]
@isuruf @scopatz does this look ready to merge now?

File: recipes/21cmFAST/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
+{% set name = "21cmFAST" %}
+{% set version = "3.0.0.dev5" %}
```
Comment (scopatz) [inline recipes/21cmFAST/meta.yaml:2]: Please package fully released versions

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml:2]: Ah. OK. That will have to wait for a little bit.

Comment (scopatz) [reply recipes/21cmFAST/meta.yaml:2]: Is there a 2.x you can package?

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml:2]: No -- version 2 was a pure C library. v3 will represent the transition to a full Python-wrapped package.

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: OK, I just updated it to point to v3.0.0

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: ... and all builds are passing, yay!

**Comment by steven-murray** [comment]
bump @ @scopatz -- I think this is ready?

File: recipes/21cmFAST/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
+{% set name = "21cmFAST" %}
+{% set version = "3.0.2.dev0" %}
```
Comment (scopatz) [inline recipes/21cmFAST/meta.yaml:2]: Is there a non-development version of the package available?

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: Agh! I thought I had updated that. It's good to go now.

File: recipes/21cmFAST/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 6e88960d134e98e4719343d853c63fc3c691438b57b2863f7834f07fae9eab4f
+
+
+build:
+  number: 0
+  script: "INC=$(CONDA_PREFIX)/include LIB=$(CONDA_PREFIX)/lib {{ PYTHON }} -m pip install . -vv"
```
Comment (scopatz) [inline recipes/21cmFAST/meta.yaml:15]: Shouldn't these be `${PREFIX}`.  `CONDA_PREFIX` is something else and we don't want a subshell

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: I'm not really sure on the difference between `PREFIX` and `CONDA_PREFIX`, but both seem to work OK, so I have switched to `PREFIX`.

Comment (scopatz) [reply recipes/21cmFAST/meta.yaml]: CONDA_PREFIX is where conda is installed. PREFIX is where the package should be installed to

File: recipes/21cmFAST/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 6e88960d134e98e4719343d853c63fc3c691438b57b2863f7834f07fae9eab4f
+
+
+build:
+  number: 0
+  script: "INC=$(PREFIX)/include LIB=$(PREFIX)/lib {{ PYTHON }} -m pip install . -vv"
```
Comment (scopatz) [inline recipes/21cmFAST/meta.yaml:15]: This should be `${PREFIX}` not `$(PREFIX)`

Comment (steven-murray) [reply recipes/21cmFAST/meta.yaml]: Ah, thanks. Done.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**